### PR TITLE
AWS-4380: Adding support for m5/m5d/r5/r5d/z1d.metal instances

### PR DESCRIPTION
**Pull request by NatahlieWang**
Adding support for m5/m5d/r5/r5d/z1d.metal instances that were announced earlier this week.

Also updating capacities for various instances; amazon had updated these values and we hadn't reflected that. [MB/s capacity, Ops/s capacity]

### CODE REVIEW

File: lib/amazon-pricing/definitions/instance-type.rb [hunk 1]
```diff
@@ -610,7 +612,7 @@ def self.get_values(json, category_type, override_price = false)
       'i3.4xlarge'  => [ 437, 16000], # EBSOptimized
       'i3.8xlarge'  => [ 875, 32500], # EBSOptimized
       'i3.16xlarge' => [1750, 65000], # EBSOptimized
-      'i3.metal'    => [1250, 64000], # EBSOptimized
+      'i3.metal'    => [1750, 65000], # EBSOptimized
       'i3p.16xlarge' => [1250, 64000], # EBSOptimized
```
Comment (stanmluke) [inline lib/amazon-pricing/definitions/instance-type.rb]: shouldn't `i3p.16xlarge` match this `i3.metal` change?

File: lib/amazon-pricing/definitions/instance-type.rb [hunk 2]
```diff
@@ -673,36 +679,37 @@ def self.get_values(json, category_type, override_price = false)
...
+      'm5a.2xlarge' => [ 265, 16000 ], #EBSOptimized peak.30min/24hrs, else [146, 8333]
       'm5a.4xlarge' => [ 265, 16000 ], #EBSOptimized
       'm5a.12xlarge' => [ 675, 30000 ], #EBSOptimized
       'm5a.24xlarge' => [ 1250, 60000 ], #EBSOptimized
-      'r5a.large' => [ 265, 16000 ], #EBSOptimized
-      'r5a.xlarge' => [ 265, 16000 ], #EBSOptimized
-      'r5a.2xlarge' => [ 265, 16000 ], #EBSOptimized
+      'r5a.large' => [ 265, 16000 ], #EBSOptimized peak.30min/24hrs, else [60, 3600]
+      'r5a.xlarge' => [ 265, 16000 ], #EBSOptimized peak.30min/24hrs, else [100, 6000]
+      'r5a.2xlarge' => [ 265, 16000 ], #EBSOptimized peak.30min/24hrs, else [146, 8333]
```
Comment (stanmluke) [inline lib/amazon-pricing/definitions/instance-type.rb]: would like to go over these (with you) PER_SEC_CAPACITIES changes, and differences between 'peak' and non-peak #'s;   we might want to confirm clients of these numbers, presumably RightSizing, will work correctly as well.

Comment (NatahlieWang) [reply lib/amazon-pricing/definitions/instance-type.rb]: Sounds good. These values were directly pulled from the AWS website; I had noticed that some of our values were off, so literally walked through each entry to confirm we're good and up to date with what they've published.

File: lib/amazon-pricing/definitions/instance-type.rb (hunk 1 above)
Comment (NatahlieWang) [reply lib/amazon-pricing/definitions/instance-type.rb]: good question; I didn't see it on their published throughput page, but I remember now we had these two having the same values as i3p.16xl is essentially similar to the i3.metal. I'll update.

File: lib/amazon-pricing/definitions/instance-type.rb (hunk 2 above)
Comment (NatahlieWang) [reply lib/amazon-pricing/definitions/instance-type.rb]: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html

**Pull request merged**